pkg/cli/cmds/k3sopts.go

```diff
@@ -60,13 +61,40 @@ func commandFromK3S(cmd cli.Command, flagOpts map[string]*K3SFlagOption) (cli.Co
 			if opt.Usage != "" {
 				strFlag.Usage = opt.Usage
 			}
-			if opt.Default != "" {
+			if opt.Default != "" || opt.OverrideDefault {
 				strFlag.Value = opt.Default
 			}
 			if opt.Hide {
 				strFlag.Hidden = true
 			}
 			flag = strFlag
+		} else if strFlag, ok := flag.(*cli.StringFlag); ok {
+			if opt.Usage != "" {
+				strFlag.Usage = opt.Usage
+			}
+			if opt.Default != "" || opt.OverrideDefault {
+				strFlag.Value = opt.Default
+			}
+			if opt.Hide {
+				strFlag.Hidden = true
+			}
+			flag = strFlag
```

Comment:
Why the duplicative cases for `*cli.StringFlag`, `*cli.IntFlag`, and `*cli.BoolFlag`?

Reply:
Because we have flags of these types, and type assertions for the non-pointer types don't work on pointers. They were being silently skipped.

Reply:
We shouldn't be mixing types. Is this left over from the spur/cli reversion to urfave/cli?

Reply:
I don't know why they vary, I just figured it was something that got broken when the cli was reverted. I can dig into it, but if it is a problem it'll need fixing on the k3s side.

Reply:
Opened an issue to follow up in k3s: https://github.com/rancher/k3s/issues/2308